### PR TITLE
Neumann BC for heat

### PR DESCRIPTION
**Pull request by voferreira**
# Description of the problem

- Lethe did not have a constant heat flux (Neumann) BC

# Description of the solution

- Implemented a heat flux boundary condition

# How Has This Been Tested?

- 2D jet case
- [x] Add application test

# Documentation

- Added the parameters to the documentation

# Future changes

- Make it space dependent as well as emissivity, h, and Tinf
- Needs some cleaning
- Make an example

### CODE REVIEW

File: source/solvers/heat_transfer_assemblers.cc [hunk 1]
```diff
@@ -421,8 +423,10 @@ HeatTransferAssemblerRobinBC<dim>::assemble_matrix(
                               const double phi_face_T_j =
                                 scratch_data.phi_face_T[f][q][j];
                               local_matrix(i, j) +=
-                                (h + 4.0 * Stefan_Boltzmann_constant *
-                                       emissivity * T_face * T_face * T_face) *
+                                (h +
+                                 4.0 * Stefan_Boltzmann_constant * emissivity *
+                                   T_face * T_face * T_face +
+                                 heat_source / T_face) *
```
Comment (blaisb) [inline source/solvers/heat_transfer_assemblers.cc:429]: Why are you dividing by T_face
A heat flux is a constant flux applied to a face. You should not divide by the temperature.
This is not temperature dependant

**Review by blaisb — CHANGES_REQUESTED** [review]
Some mistakes in the weak form and I would name that heat flux instead of heat source

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst [hunk 2]
```diff
@@ -36,6 +36,9 @@ The default parameters for ``temperature`` and ``convection-radiation`` are show
       subsection emissivity
         set Function expression = 0
       end
+      subsection heat source
```
Comment (oguevremont) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:39]:
```suggestion
      subsection heat flux
```

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst
Comment (AmishgaAlphonius) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:1]: I agree for the name, "heat flux".
Also, I think that the Neumann condition shouldn't belong in the convection-radiation subsection, as it is not convection from my understanding at l least.

**Comment by oguevremont** [comment]
Nice addition!

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst [hunk 3]
```diff
@@ -57,10 +60,10 @@ The default parameters for ``temperature`` and ``convection-radiation`` are show
...
-    * ``convection-radiation`` (Robin BC) for cooling/heating, depending on the environment temperature at the boundary ``Tinf``, with a given heat transfer coefficient ``h`` and emissivity of the boundary :math:`\mathbf{\epsilon}` following Newton's law of cooling (and heating) and Stefan-Boltzmann law of radiation. Note that the expressions for ``h``, ``Tinf`` and ``emissivity`` can be time-dependent, but the current implementation doesn't allow for space dependence (the expressions are evaluated at the origin).
+    * ``convection-radiation`` (Robin BC) for cooling/heating, depending on the environment temperature at the boundary ``Tinf``, with a given heat transfer coefficient ``h`` and ``emissivity``` of the boundary :math:`\mathbf{\epsilon}` following Newton's law of cooling (and heating) and Stefan-Boltzmann law of radiation. It is also possible to apply a heat flux using ``heat flux`` (:math:`q_0`, Neumann boundary condition). Note that the expressions for ``h``, ``Tinf``, ``emissivity``, and ``heat flux`` can be time-dependent, but the current implementation does not allow for space dependence (the expressions are evaluated at the origin).
```
Comment (AmishgaAlphonius) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:63]:
```suggestion
    * ``convection-radiation`` (Robin BC) for cooling/heating, depending on the environment temperature at the boundary ``Tinf``, with a given heat transfer coefficient ``h`` and ``emissivity``` of the boundary :math:`\mathbf{\epsilon}` following Newton's law of cooling (and heating) and Stefan-Boltzmann law of radiation. It is also possible to apply a constant heat flux using ``heat flux`` (:math:`q_0`, Neumann boundary condition). Note that the expressions for ``h``, ``Tinf``, ``emissivity``, and ``heat flux`` can be time-dependent, but the current implementation does not allow for space dependence (the expressions are evaluated at the origin).
```

**Review by AmishgaAlphonius — COMMENTED** [review]
Nice addition! I'm surprised that it wasn't there before! Even more, that I only realized it yesterday when we talked about it. :)

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst
Comment (blaisb) [reply doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:1]: I agree with this

Comment (voferreira) [reply doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:1]: Me too, but then we have two solutions: either I create another bc (wich would be a bit limiting imo), or we simply rename this one. Let me know your thoughts.

Comment (AmishgaAlphonius) [reply doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:1]: The parent subsection could be named "heat flux" and maybe the Neumann "temperature-independant heat flux" or "neumann", but then it bugs me that the Dirichlet one is not "dirichlet" and same for "robin"

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst (hunk 3 above)
Comment (AmishgaAlphonius) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:63]:
```suggestion
    * ``convection-radiation`` (Robin BC) for cooling/heating, depending on the environment temperature at the boundary ``Tinf``, with a given heat transfer coefficient ``h`` and ``emissivity``` of the boundary :math:`\mathbf{\epsilon}` following Newton's law of cooling (and heating) and Stefan-Boltzmann law of radiation. It is also possible to apply a temperature-independent heat flux using ``heat flux`` (:math:`q_0`, Neumann boundary condition). Note that the expressions for ``h``, ``Tinf``, ``emissivity``, and ``heat flux`` can be time-dependent, but the current implementation does not allow for space dependence (the expressions are evaluated at the origin).
```

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst
Comment (voferreira) [reply doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:1]: What do you think of "q0"? It would be similar to "Tinf". And then the section could be "heat flux"

Comment (voferreira) [reply doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:1]: @blaisb Can we agree on this before I change all files?

**Comment by AmishgaAlphonius** [comment]
Also I just realized, but an application-test can also be added for this new BC. :)

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+#---------------------------------------------------
+
+subsection simulation control
+  set method           = bdf1
+  set time end         = 0.01
+  set time step        = 0.01
+  set adapt            = false
+  set output name      = heat_flux
+  set output frequency = 1
+end
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:17]: Why not instead add an application state with a steady-state solution which is analytical? This would make it easier to validate that the code is fully working well no?

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+#---------------------------------------------------
+# Post-Processing
+#---------------------------------------------------
+
+subsection post-processing
+  set verbosity                        = verbose
+  set calculate temperature statistics = true
+end
+
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:35]: Not sure this is necessary in this test, but if you want to keep iot we can

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst [hunk 6]
```diff
@@ -37,6 +37,13 @@ The default parameters for ``temperature`` and ``convection-radiation`` are show
         set Function expression = 0
       end
     end
+    subsection bc 2
+      set id     = 2
+      set type   = heat-flux
+      subsection q0
+        set Function expression = 0
+      end
```
Comment (blaisb) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:45]: I'm not sure q0 is the right name here. why not just heat-flux?

File: include/solvers/heat_transfer_assemblers.h [hunk 7]
```diff
@@ -217,6 +217,52 @@ class HeatTransferAssemblerRobinBC : public HeatTransferAssemblerBase<dim>
   const BoundaryConditions::HTBoundaryConditions<dim> &boundary_conditions_ht;
 };
 
+/**
+ * @brief Class that assembles prescribed heat flux Neumann boundary condition for the heat transfer solver.
```
Comment (blaisb) [inline include/solvers/heat_transfer_assemblers.h:221]:
```suggestion
 * @brief Assembles prescribed heat flux Neumann boundary condition for the heat transfer solver.
```

File: source/solvers/heat_transfer.cc [hunk 8]
```diff
@@ -348,6 +348,11 @@ HeatTransfer<dim>::setup_assemblers()
     std::make_shared<HeatTransferAssemblerRobinBC<dim>>(
       this->simulation_control, simulation_parameters.boundary_conditions_ht));
 
+  // Neumann boundary condition
```
Comment (blaisb) [inline source/solvers/heat_transfer.cc:351]: The best things to do for these BCs would be to add the assemblers only if the BC type exists, but maybe that can be another PR.

**Review by blaisb — CHANGES_REQUESTED** [review]
Just a few comments

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm (hunk 4 above)
Comment (voferreira) [reply applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:17]: It is better, indeed. Just made it.

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst (hunk 6 above)
Comment (voferreira) [reply doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:45]: I just changed it to "value", to follow what is being used for the Dirichlet BC.

File: source/solvers/heat_transfer.cc (hunk 8 above)
Comment (voferreira) [reply source/solvers/heat_transfer.cc:351]: I will do it in this one.

Comment (voferreira) [reply source/solvers/heat_transfer.cc:357]: I choose to define a "has_type_bc" boolean to be "true" when the parameter is called but I don't know if this is the cleanest approach.

Comment (blaisb) [reply source/solvers/heat_transfer.cc:357]: Well that's not a bad approach. It's easy to understand and boundary conditions don't change type during simulations so it makes sense.

File: source/solvers/heat_transfer_assemblers.cc [hunk 9]
```diff
@@ -486,7 +489,8 @@ HeatTransferAssemblerRobinBC<dim>::assemble_rhs(
                             (h * (T_face - T_inf) +
                              Stefan_Boltzmann_constant * emissivity *
                                (T_face * T_face * T_face * T_face -
-                                T_inf * T_inf * T_inf * T_inf)) *
+                                T_inf * T_inf * T_inf * T_inf) +
+                             heat_flux_bc) *
```
Comment (blaisb) [inline source/solvers/heat_transfer_assemblers.cc:493]:
```suggestion
                             ) *
```
You added a seperate assembler, so this heat flux should not be in the Robin BC anymore

File: source/solvers/heat_transfer_assemblers.cc [hunk 10]
```diff
@@ -465,6 +465,9 @@ HeatTransferAssemblerRobinBC<dim>::assemble_rhs(
             this->boundary_conditions_ht.Tinf[i_bc]->value(Point<dim>());
           const double emissivity =
             this->boundary_conditions_ht.emissivity[i_bc]->value(Point<dim>());
+          const double heat_flux_bc =
+            this->boundary_conditions_ht.heat_flux_bc[i_bc]->value(
+              Point<dim>());
```
Comment (blaisb) [inline source/solvers/heat_transfer_assemblers.cc:470]:
```suggestion

```
The heat flux BC is seperate from convection-diffusion, so you should remove it from there.

**Review by blaisb — CHANGES_REQUESTED** [review]
One tiny last change (you forgot the flux in the Robin boundary condition)

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst [hunk 11]
```diff
@@ -57,11 +64,21 @@ The default parameters for ``temperature`` and ``convection-radiation`` are show
...
+.. math::
+    \frac{ \partial T}{\partial \mathbf{n}} = q_0
+
+
+.. note::
+
+  Note that the expressions for ``h``, ``Tinf``, ``emissivity``, and ``heat-flux`` can be time-dependent, but the current implementation does not allow for space dependence (the expressions are evaluated at the origin).
+
 
 where :math:`\mathbf{\sigma}` is the Stefan-Boltzmann constant.
```
Comment (AmishgaAlphonius) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:83]: The indentations were coming out wrong.

```suggestion
    .. math::
        \frac{ \partial T}{\partial \mathbf{n}} = h (T - T_{inf}) + \epsilon \sigma (T^4 - T_{inf}^4)

    where :math:`\mathbf{\sigma}` is the Stefan-Boltzmann constant.

    * ``heat-flux`` (:math:`q_0`, Neumann BC) to impose a heat flux of a given ``value`` at the boundary.

    .. math::
        \frac{ \partial T}{\partial \mathbf{n}} = q_0


    .. note::

      Note that the expressions for ``h``, ``Tinf``, ``emissivity``, and ``heat-flux`` can be time-dependent, but the current implementation does not allow for space dependence (the expressions are evaluated at the origin).
```

File: include/solvers/heat_transfer_assemblers.h [hunk 12]
```diff
@@ -217,6 +217,52 @@ class HeatTransferAssemblerRobinBC : public HeatTransferAssemblerBase<dim>
...
 };
 
+/**
+ * @brief Assembles prescribed heat flux Neumann boundary condition for the heat transfer solver.
+ *
+ * @tparam dim An integer that denotes the number of spatial dimensions
+ *
+ * @param simulation_control Shared pointer of the SimulationControl object
+ * controlling the current simulation
+ * @param p_boundary_conditions_ht HTBoundaryConditions object that hold
```
Comment (AmishgaAlphonius) [inline include/solvers/heat_transfer_assemblers.h:227]:
```suggestion
 * @param p_boundary_conditions_ht HTBoundaryConditions object that holds
```

**Comment by voferreira** [comment]
@blaisb Thanks, my bad.

File: source/solvers/heat_transfer_assemblers.cc [hunk 13]
```diff
@@ -500,6 +500,63 @@ template class HeatTransferAssemblerRobinBC<2>;
...
+                {
+                  for (unsigned int q = 0; q < scratch_data.n_faces_q_points;
+                       ++q)
+                    {
+                      const double JxW = scratch_data.face_JxW[f][q];
+                      for (unsigned int i = 0; i < scratch_data.n_dofs; ++i)
+                        {
+                          const double phi_face_T_i =
+                            scratch_data.phi_face_T[f][q][i];
+                          local_rhs(i) += phi_face_T_i * heat_flux_bc * JxW;
```
Comment (AmishgaAlphonius) [inline source/solvers/heat_transfer_assemblers.cc:547]: Shouldn't it be a negative here, and in the parameters you impose the positive value? (application test's value must be changed)
```suggestion
                          local_rhs(i) -= phi_face_T_i * heat_flux_bc * JxW;
```

**Review by AmishgaAlphonius — COMMENTED** [review]
A few comments and suggestions :)

File: source/solvers/heat_transfer_assemblers.cc (hunk 13 above)
Comment (blaisb) [reply source/solvers/heat_transfer_assemblers.cc:547]: Well it depends right. Because the boundary condition when you have a flux is  -k * grad(T) = q. Consequently, most likley there is (-)*(-) being lumped here. The analytical solution problem will be able to tell you what it is. A positive flux means heat is going out (flux is aligned with the outward normal) and a negative flux means the heat is going in.

Comment (blaisb) [reply source/solvers/heat_transfer_assemblers.cc:547]: Using this definition (positive flux means alligned with outward normal and negative means agaisnt it), it means that currently there is indeed a sign mistake. You are specifying a flux of -1000smthing in your prm file. Consequently, we would expect that this means that energy is getting inside of the system (it is not aligned with the outward normal, but on the opposite side). So I think I would clear this out in the documentation (mention that the flux is aligned with the unit normal going out) and that a flux getting in thus has to be negative and a flux going out thus has the be positive. This would make things a lot clearer and would prevent this type of confusion again. It's the issue when you don't have Robin BCs, the convention is not trivial to define.

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+set dimension = 2
+
+#---------------------------------------------------
+# Simulation Control
+#---------------------------------------------------
+
+subsection simulation control
+  set method           = steady
+  set output name      = heat_flux
+  set output frequency = 1
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:13]:
```suggestion
  set output frequency = 0
```

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+
+subsection initial conditions
+  set type = nodal
+  subsection uvwp
+    set Function expression = 0; 0; 0
+  end
+  subsection temperature
+    set Function expression = 1000
+  end
+end
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:36]:
```suggestion
```
This is not necessary here since this is a steady problem

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+#---------------------------------------------------
+# Analytical Solution
+#---------------------------------------------------
+
+subsection analytical solution
+  set enable    = true
+  set verbosity = verbose
+  subsection uvwp
+    set Function expression = 0 ; 0 ; 0
+  end
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:104]:
```suggestion
```

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+end
+
+#---------------------------------------------------
+# FEM
+#---------------------------------------------------
+
+subsection FEM
+  set velocity order = 1
+  set pressure order = 1
+end
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:117]:
```suggestion
```

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+
+#---------------------------------------------------
+# Non-Linear Solver Control
+#---------------------------------------------------
+
+subsection non-linear solver
+  subsection heat transfer
+    set tolerance      = 1e-8
+    set max iterations = 100
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:126]:
```suggestion
```

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 19]
```diff
@@ -0,0 +1,163 @@
...
+    set verbosity      = quiet
+  end
+  subsection fluid dynamics
+    set tolerance      = 1e-8
+    set max iterations = 100
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:131]:
```suggestion
```

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 20]
```diff
@@ -0,0 +1,163 @@
...
+    set verbosity                             = quiet
+    set method                                = gmres
+    set relative residual                     = 1e-3
+    set minimum residual                      = 1e-8
+    set preconditioner                        = ilu
+    set ilu preconditioner fill               = 0
+    set ilu preconditioner absolute tolerance = 1e-12
+    set ilu preconditioner relative tolerance = 1.00
+    set max krylov vectors                    = 200
+  end
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:151]:
```suggestion

```

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 21]
```diff
@@ -0,0 +1,163 @@
...
+  subsection heat transfer
+    set verbosity                             = quiet
+    set method                                = gmres
+    set relative residual                     = 1e-3
+    set minimum residual                      = 1e-8
+    set preconditioner                        = ilu
+    set ilu preconditioner fill               = 0
+    set ilu preconditioner absolute tolerance = 1e-12
+    set ilu preconditioner relative tolerance = 1.00
+    set max krylov vectors                    = 200
```
Comment (blaisb) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:161]:
```suggestion
```

You can use default values here

**Review by blaisb — CHANGES_REQUESTED** [review]
Some suggestions for the prm file (use default values will make your life easier)

File: source/solvers/heat_transfer_assemblers.cc (hunk 13 above)
Comment (voferreira) [reply source/solvers/heat_transfer_assemblers.cc:547]: I made it + because I felt it to be more intuitive for the user to do positive if you say heat is going in and negative if heat is going out of the system. Can I just add this explanation to the documentation or you would rather have me changing to take as reference the normal vector (and still add this to the docs)?

Comment (blaisb) [reply source/solvers/heat_transfer_assemblers.cc:547]: Well it depends. I guess to me it makes more sense that negative is going in (in a vector sense), but we don't have a lot of things like that in the code so as long as we are clear, it's ok. I would prefer (-) = in and (+) = out, but if other people feel strongly about the other way, I can be flexible. I just prefer the mathematical definition of it lets say

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 22]
```diff
@@ -0,0 +1,125 @@
...
+#---------------------------------------------------
+
+subsection multiphysics
+  set heat transfer = true
+end
+
+#---------------------------------------------------
+# Initial condition
+#---------------------------------------------------
+
```
Comment (oguevremont) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:27]:
```suggestion
```

File: applications_tests/lethe-fluid/heat_transfer_heat-flux.prm [hunk 23]
```diff
@@ -0,0 +1,125 @@
...
+  set verbosity = verbose
+  subsection temperature
+    set Function expression = 1000 - 100*x
+  end
+end
+
+#---------------------------------------------------
+# FEM
+#---------------------------------------------------
+
```
Comment (oguevremont) [inline applications_tests/lethe-fluid/heat_transfer_heat-flux.prm:98]:
```suggestion
```

File: source/solvers/heat_transfer_assemblers.cc (hunk 13 above)
Comment (oguevremont) [reply source/solvers/heat_transfer_assemblers.cc:547]: I would also prefer (-) = in, but whatever you choose leaving a note that mentions it would be welcome and could save some time for users.

Comment (voferreira) [reply source/solvers/heat_transfer_assemblers.cc:547]: (-) it is

**Review by blaisb — APPROVED** [review]

**Review by blaisb — APPROVED** [review]

**Review by oguevremont — APPROVED** [review]

**Comment by voferreira** [comment]
As prof @blaisb and I have discussed Today, I will add the imposed heat flux as a term to the Robin BC again because it is more versatile. This will help with the further implementation of the space-dependent Robin BC.

File: include/core/boundary_conditions.h [hunk 24]
```diff
@@ -443,7 +443,9 @@ namespace BoundaryConditions
    * convective heat transfer coefficient and "Tinf" is the
    * environment temperature at the boundary, "emissivity" is the emissivity
    * coefficient, and "Stefan-Boltzmann constant" is the Stefan-Boltzmann
-   * constant = 5.6703*10-8 (W.m-2.K-4)
+   * constant = 5.6703*10-8 (W.m-2.K-4). It also possible to impose a heat flux
```
Comment (AmishgaAlphonius) [inline include/core/boundary_conditions.h:446]:
```suggestion
   * constant = 5.6703*10-8 \f$(W.m^{-2}.K^{-4})\f$. It is also possible to impose a heat flux
```

**Review by AmishgaAlphonius — APPROVED** [review]
One small comment :)

File: doc/source/parameters/cfd/boundary_conditions_multiphysics.rst [hunk 25]
```diff
@@ -57,13 +61,20 @@ The default parameters for ``temperature`` and ``convection-radiation`` are show
...
     * ``noflux`` (default) so that there is no heat transfer boundary condition,
     * ``temperature`` (Dirichlet BC), to impose a given temperature ``value`` at the boundary,
-    * ``convection-radiation`` (Robin BC) for cooling/heating, depending on the environment temperature at the boundary ``Tinf``, with a given heat transfer coefficient ``h`` and emissivity of the boundary :math:`\mathbf{\epsilon}` following Newton's law of cooling (and heating) and Stefan-Boltzmann law of radiation. Note that the expressions for ``h``, ``Tinf`` and ``emissivity`` can be time-dependent, but the current implementation doesn't allow for space dependence (the expressions are evaluated at the origin).
+    * ``convection-radiation`` (Robin BC) for cooling/heating, depending on the environment temperature at the boundary ``Tinf``, with a given heat transfer coefficient ``h`` and ``emissivity`` of the boundary :math:`\mathbf{\epsilon}` following Newton's law of cooling (and heating) and Stefan-Boltzmann law of radiation. It is also possible to impose a given heat flux (:math:`q_0`) by using the parameter ``heat_flux``. This BC can be represented by:
```
Comment (blaisb) [inline doc/source/parameters/cfd/boundary_conditions_multiphysics.rst:64]: maybe we would want to rename convection-radiation convection-radiation-flux? If so. I would add a CHANGELOG. But if you want, we can do this in a next PR... This way we can merge this one now

**Comment by blaisb** [comment]
@voferreira write to me on slack when ready for merge :)

**Comment by blaisb** [comment]
can you please fix CI @voferreira
